### DISCUSSION TITLE
Can't send/receive message in same thread

### DISCUSSION THREAD

**Discussion by jholloway7** [mido/mido#420] [Q&A]
Should this work?  

```py
import mido

def test_send_receive_same_thread():
    o = mido.open_output('MIDO-TEST', virtual=True)
    i = mido.open_input('MIDO-TEST', virtual=True)

    msg = mido.Message('clock')

    o.send(msg)

    result = list(i.iter_pending())

    assert len(result) == 1

def test_send_receive_same_thread_ioport():
    msg = mido.Message('clock')

    io = mido.open_ioport('MIDO-TEST', virtual=True)

    io.send(msg)

    result = list(io.iter_pending())

    assert len(result) == 1


```

Thanks

**Comment by rdoursenaud** [comment]
If this is still current, could you please describe a little bit more what you are trying to achieve. The code snippet is unclear to me.

**Comment by jholloway7** [comment]
I think we can close this. My use case here was probably only valid for unit testing.  I may have simplified my code too much for it to make sense.  Ultimately I was writing a midi clock implementation and wanting to test that I was generating the pulses correctly without having a real bus.  I thought I might be able to do it by wiring up two virtual ports as a "mock" channel that I could pass to my midi clock engine and validate that the messages were arriving as expected.  I haven't touched the code in a while, but I was able to create a slight indirection on top of mido that achieved the same thing.
